src/tea_json.res

```diff
@@ -222,35 +222,53 @@ module Decoder = {
 
   let map = (mapper, Decoder(decoder1)) => Decoder(
     value => {
-      open Tea_result
       switch decoder1(value) {
       | Ok(v1) => Ok(mapper(v1))
       | Error(e) => Error("map " ++ e)
       }
     },
   )
 
+let error = x =>
+  switch x {
+  | Ok(_) => None
+  | Error(e) => Some(e)
+  }
+
+let error_of_first = (fst, x) =>
+  switch x {
+  | Error(e) => Some(e)
+  | Ok(_) => error(fst)
+  }
+
+
+
   let map2 = (mapper, Decoder(decoder1), Decoder(decoder2)) => Decoder(
     value => {
-      open Tea_result
       switch (decoder1(value), decoder2(value)) {
       | (Ok(v1), Ok(v2)) => Ok(mapper(v1, v2))
       | (e1, e2) =>
-        switch Tea_result.error_of_first(e1, e2) {
+        switch error_of_first(e1, e2) {
         | None => failwith("Impossible case")
         | Some(e) => Error("map2 -> " ++ e)
         }
       }
     },
   )
 
+  let first = (fst, x) =>
```

Comment:
I think this would be clearer if it was inlined into where is used

Reply:
not sure if I understood this correctly but I made the changes if it is wrong I will fix it.